### PR TITLE
feat(dockerslim): dockerslim sample task

### PR DESCRIPTION
**Pull request by faust64**
# Changes

Introduces a new sample task, using dockerslim builder image, building container images.

See https://github.com/docker-slim/docker-slim
Following up on some StackOverflow question that caught my attention - https://stackoverflow.com/questions/70193930/how-to-create-tekton-task-for-building-a-docker-slim

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you review them:

- [X] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [X] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [X] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [~] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [X] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [X] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [X] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [X] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [X] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [X] mandatory `spec.description` follows the convention

(I duplicated the docker-build / dind sample task, with some minor modifications, ... should be fine)

### CODE REVIEW

File: task/dockerslim-build/0.1/tests/run.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: dockerslim-test-pipeline-run
+spec:
+  pipelineRef:
+    name: dockerslim-test-pipeline
+  workspaces:
+  - name: shared-workspace
+    persistentvolumeclaim:
```
Comment (vinamra28) [inline task/dockerslim-build/0.1/tests/run.yaml:45]: nit
```suggestion
    persistentVolumeClaim:
```

File: task/dockerslim-build/0.1/OWNERS [hunk 2]
```diff
@@ -0,0 +1,2 @@
+approvers:
+reviewers:
```
Comment (vinamra28) [inline task/dockerslim-build/0.1/OWNERS:2]: can we omit this file if nobody is the owner of this task?

**Review by vinamra28 — COMMENTED** [review]
/ok-to-test

File: task/dockerslim-build/0.1/OWNERS (hunk 2 above)
Comment (faust64) [reply task/dockerslim-build/0.1/OWNERS:2]: How does it work? Isn't this file supposed to list the usernames for organization members that did review / approve that task folder getting merged?
For sure, I can remove that one, if it's not mandatory.

**Review by vinamra28 — COMMENTED** [review]
PR looks good :+1: 
can you please squash the commits as well?

**Review by vinamra28 — APPROVED** [review]
/meow

**Pull request merged**